src/common/sockets.cc

```diff
@@ -606,11 +606,15 @@ int32_t tcptoread(int sock, void *buff, uint32_t leng, int msecto) {
 	while (rcvd < leng) {
 		pfd.revents = 0;
 		if (tcppoll(pfd, msecto) < 0) {
+			tcpsetlasterror(TCPETIMEDOUT);
 			return -1;
 		}
 		if (pfd.revents & POLLIN) {
 			i = tcprecv(sock, ((uint8_t *)buff) + rcvd, leng - rcvd, 0);
 			if (i == 0 || (i < 0 && tcpgetlasterror() != TCPEAGAIN)) {
+				if (i == 0) {
+					tcpsetlasterror(TCPETIMEDOUT);
+				}
```

Comment:
`tcprecv` seems to be just a wrapper function for `recv`

From `man recv`

```
RETURN VALUE
       Upon successful completion, recv() shall return the length of  the  mes‐
       sage  in bytes. If no messages are available to be received and the peer
       has performed an orderly shutdown, recv() shall return 0. Otherwise,  -1
       shall be returned and errno set to indicate the error.
```

Is it then accurate to set this as ETIMEDOUT as the error value? It did not timeout, it just responded with nothing.

Maybe it should EAGAIN (but this might have consequences in other parts of the code if it's checking that). Have a look at `man errno` and see what error may best fit in this situation.

Reply:
At the end I decided it to be `TCPNORESONSE` which is a definition of `EIO` generic _input/ouput_ error. This way I show a more generic reason. And also not affecting the common library as you suggest. Please check my lastest commit for this.